### ISSUE TITLE
Support automated dataset-id and dataset-version fetching in meta-add

### ISSUE THREAD

**Issue by christian-monch** [datalad/datalad-metalad#114]
Currently, meta-add requires to specify a dataset-ID and a dataset-version. That is useful if the metadata was, for example, dumped from a metadata store "a" and should be added to the different metadata store "b". 

But it is also cumbersome to add the two values every time. We should therefore add a switch that allows to read them from a locally available dataset.

**Comment by bpoldrack** [comment]
This:

> Or: if issue #114 is implemented 

is a confusing self-reference. Not sure, what to make of this.

> We should therefore add a switch that allows to read them from a locally available dataset.

Just like with other commands, I'd say the other way around: The default for a datalad command should be to operate on "the dataset". Options may allow for pointing elsewhere as a source (where we might still be able to automatically get ID and version) and to explicitly pass in ID/version if not available.

**Comment by christian-monch** [comment]
> 
> is a confusing self-reference. Not sure, what to make of this.
> 

Thx, fixed it.

> > We should therefore add a switch that allows to read them from a locally available dataset.
> 
> Just like with other commands, I'd say the other way around: The default for a datalad command should be to operate on "the dataset". Options may allow for pointing elsewhere as a source (where we might still be able to automatically get ID and version) and to explicitly pass in ID/version if not available.


I agree. Will stop "pushing/keeping" the separation between dataset and metadata store for the dataset on the UI. See issue #119.

This should also allow us to automatically read required information from the datasets involved in an operation

**Comment by christian-monch** [comment]
@bpoldrack: as far as I can see, implementing this issue (and #119) would require a local instance of a dataset in order to add metadata to it. This is in contrast to the current situation: you do not need the dataset itself, just the extracted metadata record and a local git-repository, which holds the metadata. Do you see any merits in such a configuration?

**Comment by bpoldrack** [comment]
> as far as I can see, implementing this issue (and #119) would require a local instance of a dataset in order to add metadata to it.
> This is in contrast to the current situation: you do not need the dataset itself, just the extracted metadata record and a local git-repository, which holds the metadata

Yes, we only can add metadata to a dataset. A local git repository is a dataset from datalad's POV. I see your point in "we could store the metadata in a different dataset (that may not have an ID)", but it makes little sense to me. Why would I want the metadata in a dataset ("another local git repo" in your wording) to actually be the metadata of another, unrelated dataset? Somewhat contradicts the "meta" in "metadata" and the idea that any repository is a dataset. I don't see why one would want this. What exactly can be achieved that way, that couldn't be achieved otherwise? To me this complicates/confuses everything w/o achieving anything. If I need a particular clone to only hold the metadata for whatever reason, I can still have a bare clone without any data content.
